src/package-vsix.js

```diff
@@ -0,0 +1,117 @@
+    choices: ['latest', 'next']
+}).demandOption('tag').argv;
+
+const repository = {
+    "type": "git",
+    "url": "https://github.com/theia-ide/vscode-builtin-extensions"
+};
+
+// bump to publish
+let version = '1.39.1';
+
+(async () => {
+
+    const bin = await run('yarn', ['bin'], root());
+    const vsce = bin.trim() + '/vsce';
+
+    if (tag === 'next') {
+        const shortRevision = (await run('git', ['rev-parse', '--short', 'HEAD'], vscode())).trim();
+        const [, minor] = version.split('.');
+        version = `0.${Number(minor) + 1}.0-next.${shortRevision}`;
+    }
+    const result = [];
+
+    // typescript-language-features ext needs "extensions/node_modules" content
+    // and a bit of massaging to work as standalone .vsix. Basically replace this:
+    //      "vscode.typescript-language-features",["..","node_modules"]
+    // with this:
+    //      "vscode.typescript-language-features",[".","deps"]
+    // in the compiled extension, so that the TS LS will be packaged and found at runtime
+    const extensionsNodeModulesPath = extensions('node_modules');
+    const tsLangFeaturesNMPath = extensions('typescript-language-features');
+    if (fs.existsSync(extensionsNodeModulesPath) && fs.existsSync(tsLangFeaturesNMPath)) {
+        await fs.copy(extensionsNodeModulesPath, tsLangFeaturesNMPath + '/deps');
+        console.log('Copying node_modules under typescript-language-features');
+
+        const extjs = extensions('typescript-language-features', 'dist', 'extension.js');
+        const orig = '"vscode.typescript-language-features",["..","node_modules"]';
+        const modified = '"vscode.typescript-language-features",[".","deps"]';
+        const extjsOrigContent = fs.readFileSync(extjs, 'utf-8');
+        if (extjsOrigContent.includes(orig)) {
+            console.log('TS language compliled extension is original - patching')
+            // const extjsNewContent = extjsOrigContent.replace(orig, modified);
+            fs.writeFileSync(extjs, extjsOrigContent.replace(orig, modified), 'utf-8');
+        }
+        else {
+            console.log('TS language extension is already patched')
+        }        
+    }
+
+    if (!fs.existsSync(out())) {
+        await fs.mkdir(out());
+    }
+
+    for (const extension of fs.readdirSync(extensions())) {
+        // console.log(`extension: ${extension}`);
+        if (extension.includes('node_modules')) {
+            continue;
+        }
+        const pckPath = extensions(extension, 'package.json');
+        if (!fs.existsSync(pckPath)) {
+            continue;
+        }
+
+        const originalContent = fs.readFileSync(pckPath, 'utf-8');
+        const pck = JSON.parse(originalContent);
+        // warning: do not meddle with pck.publisher - it's part of the 
+        // extension id and used in places to access some extensions
+        // pck.publisher = "theia-ide";
+        pck.repository = repository;
+        pck.version = version;
```

Comment:
some package are private, for npm publishing i had to delete `private` option to make it work. Does vsce care about it as well?

Reply:
No, it does not care as far as I can tell